File: Assignment__22/Question5.java
//Accept N numbers from user and accept one another number as NO return frequency of NO form it.

import java.util.Scanner;

class ArrayX
{
    public int Arr[];
    public int iSize;

    public ArrayX(int iNo)
    {
        iSize = iNo;

        Arr = new int [iSize];    //Need to remember
    }

    public void Accept()
    {
        Scanner sobj1 = new Scanner(System.in);

        int iCnt = 0;

        System.out.println("Enter the numbers : ");

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            Arr[iCnt] = sobj1.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("The elements of the array are : ");

        int iCnt = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            System.out.println(Arr[iCnt]);
        }
    }

    public int CheckNumberFreq(int iNum)
    {
        int iCnt = 0;
        int iCount = 0;

        for(iCnt = 0; iCnt < Arr.length; iCnt++)
        {
            if(Arr[iCnt] == iNum)
            {
                iCount++;
            }   
        }
        return iCount;
    }
}

class Question5
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the number of elements : ");
        int iValue1 = sobj.nextInt();

        System.out.println("Enter the number that shoud be found : ");
        int iValue2 = sobj.nextInt();

        int iResult = 0;

        ArrayX aobj = new ArrayX(iValue1);
        aobj.Accept();
        aobj.Display();

        iResult = aobj.CheckNumberFreq(iValue2);

        System.out.println(iResult+" is the frequency of input number in the given array.");

        aobj = null;
        sobj = null;

        System.gc();
    }

}

File: Assignment__22/Question5.c
//Accept N numbers from user and accept one another number as NO return frequency of NO form it.
    
#include<stdio.h>
#include<stdlib.h>

int Count(int Arr[], int iSize, int iNo)
{
    int iCnt = 0;c

    int iCount = 0;

    for(iCnt = 0; iCnt < iSize; iCnt++)    
    {
        if(Arr[iCnt] == iNo)
        {
            iCount = iCount + 1;
        }
    }

    return iCount;
}

int main()
{
    int iLength = 0;
    int iValue = 0;
    int iCnt = 0;
    int *ptr = NULL;
    int iResult = 0;

    printf("Enter the number of elements : ");
    scanf("%d", &iLength);

    ptr = (int *)malloc(iLength * sizeof(int));
    
    if(ptr == NULL)      
     {
        printf("Unable to allocate the memoey\n");
        return -1;
    }

    printf("Enter the elements : \n");

    for(iCnt = 0; iCnt < iLength; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    printf("Enter the number that should be count : \n");
    scanf("%d", &iValue);

    iResult = Count(ptr,iLength,iValue);
    printf("The frequency of given number in the given array is : %d", iResult);

    free(ptr);

    return 0;

}
